sdk/scheduler/src/main/java/com/mesosphere/sdk/scheduler/plan/Status.java

```diff
@@ -49,6 +44,17 @@
     /**
      * Execution has completed.
      */
-    COMPLETE
+    COMPLETE,
+
+    //TODO: make enum implement an interface, forbid step status to be set to WAITING or IN_PROGRESS
+    /**
+     * RETURN VALUE: only returned when execution has been interrupted.
+     */
+    WAITING,
 
+    /**
+     * RETURN VALUE: only returned by Phase and Plan getStatus, to state that at lease one child is complete and at
+     * least one child is in progress (either PENDING or prepared PREPARED).
+     */
+    IN_PROGRESS
```

Comment:
Let's avoid ordering changes on enums.  It's potentially dangerous for upgrades.  It doesn't matter since we don't persist these right now, but let's get in the habit of just appending to the enum list OK?